kbase-extension/static/kbase/js/widgets/kbaseTabs.js

```diff
@@ -64,167 +64,208 @@ define(['kbwidget', 'bootstrap', 'jquery', 'kbaseDeletePrompt'], (
         },
 
         appendUI: function ($elem, tabs) {
-            if (tabs == undefined) {
+            if (typeof tabs === 'undefined') {
                 tabs = this.options.tabs;
             }
 
-            const $block = $('<div></div>').addClass('tabbable');
-            const $tabs = $('<div></div>')
-                .addClass('tab-content')
-                .attr('id', 'tabs-content')
-                .css('height', this.tabsHeight());
-            const $nav = $('<ul role="tablist"></ul>')
-                .addClass('nav nav-tabs')
-                .attr('id', 'tabs-nav');
-            $block.append($nav).append($tabs);
-            /*if (this.options.tabPosition == 'top') {
-                $block.addClass('tabs-above');
-                $block.append($nav).append($tabs);
-            }
-            else if (this.options.tabPosition == 'bottom') {
-                $block.addClass('tabs-below');
-                $block.append($tabs).append($nav);
-            }
-            else if (this.options.tabPosition == 'left') {
-                $block.addClass('tabs-left');
-                $block.append($nav).append($tabs);
-            }
-            else if (this.options.tabPosition == 'right') {
-                $block.addClass('tabs-right');
-                $block.append($tabs).append($nav);
-            }*/
+            const $tabset = $('<div>').addClass('tabbable');
 
-            this._rewireIds($block, this);
+            const $tabs = $('<div>').addClass('tab-content').attr('id', 'tabs-content');
 
-            $elem.append($block);
+            const $nav = $('<ul role="tablist" />').addClass('nav nav-tabs').attr('id', 'tabs-nav');
 
-            if (tabs) {
-                $.each(
-                    tabs,
-                    $.proxy(function (idx, tab) {
-                        this.addTab(tab);
+            $tabset.append($nav).append($tabs);
+
+            this._rewireIds($tabset, this);
+
+            $elem.append($tabset);
+
+            if (tabs && tabs.length > 0) {
+                // defaults to first tab being active if none are set as active by default.
+                if (
+                    !tabs.some((tab) => {
+                        return tab.show;
+                    })
+                ) {
+                    tabs[0].show = true;
+                }
+                tabs.forEach((tab) => {
+                    this.addTab(tab);
+                });
+            }
+        },
+
+        renderDeleteTabButton: function (tab) {
+            return $('<button>')
+                .addClass('btn btn-default btn-xs')
```

Comment:
can you give this button a unique class (`btn-delete`?) and add all the css there, rather than having it hardcoded here?

Reply:
Not really hard coded is it? These are "standard" bootstrap button classes, meant to be combined like this.
Maybe I'm not understanding your point.